oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryOptions.java

```diff
@@ -106,6 +110,14 @@ public QueryOptions(QueryOptions defaultValues) {
             t.read(']');
             prefetch = list;
         }
+        x = map.get("insecureResultSize");
+        if (x != null) {
```

Comment:
I think the following should work:

    insecureResultSize = Boolean.parseBoolean(map.get("insecureResultSize"))